### PR TITLE
NOMERGE Test app

### PR DESCRIPTION
**Pull request by dbenamydd**
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

File: test-apps/basic/main.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package main
```
Comment (nsrip-dd) [inline test-apps/basic/main.go:1]: We'll probably want the test apps to live in the `internal/` directory, like `internal/test-apps` to be clear that these aren't for public consumption :)

**Review by nsrip-dd — COMMENTED** [review]
Really nice! I like this simple approach and think we could get a lot of mileage out of it. A few questions:

* Does this deploy automatically right now, or do we manually trigger it on GitLab?
* If it's automatic and somebody is pushing a bunch of small commits to a branch in short succession, will the new commits cancel the old job?
* Can we do more than one at a time with this approach? I'm thinking of something like, maybe we want to start up every app for commits to main.

For reference, GitLab does support running multiple containers as part of a job (they call it [services](https://docs.gitlab.com/ee/ci/services/#services)), if we wanted to incorporate an external service in the future. But, I think I'd only want to do that if we really can't get what we want done with self contained apps plus a shell script :)

**Comment by dbenamydd** [comment]
Thanks for taking a look!

> Does this deploy automatically right now, or do we manually trigger it on GitLab?

Automatic 1h run of the "basic" test app. This is controlled by the `trigger_internal_build` job in .gitlab-ci.yml in this PR.

> If it's automatic and somebody is pushing a bunch of small commits to a branch in short succession, will the new commits cancel the old job?

Not atm.

> Can we do more than one at a time with this approach? I'm thinking of something like, maybe we want to start up every app for commits to main.

I wonder if we can have multiple dd-trace-go jobs which each trigger their own downstream pipeline. That would keep the interface between the 2 repos relatively simple. If that doesn't work and that functionality is important, we'll figure out how to do something fancier. Let's come back to it once we have a second test app?

> For reference, GitLab does support running multiple containers as part of a job (they call it [services](https://docs.gitlab.com/ee/ci/services/#services)), if we wanted to incorporate an external service in the future. But, I think I'd only want to do that if we really can't get what we want done with self contained apps plus a shell script :)

Agreed, we should see how far we can get with the simple approach.

**Comment by nsrip-dd** [comment]
> Automatic 1h run of the "basic" test app. This is controlled by the trigger_internal_build job in .gitlab-ci.yml in this PR.

Got it, thanks. I'm mildly concerned that if we have a rapid sequence of pushes to the same PR, then we'll end up with unneeded jobs using up resources, where only the most recent one is actually testing the current PR code. But, I think we could address that if it becomes a problem.

> I wonder if we can have multiple dd-trace-go jobs which each trigger their own downstream pipeline. That would keep the interface between the 2 repos relatively simple. If that doesn't work and that functionality is important, we'll figure out how to do something fancier. Let's come back to it once we have a second test app?

Sounds good to me.

**Comment by nsrip-dd** [comment]
One more thing I noticed, the app's profiles gets tagged as `service:gitlab-runner`:

Screen Shot 2022-11-18 at 2 40 09 PM

The Go profiler always adds a `service` tag to the profile uploads, defaulting to the name of the program. So it seems like something is overriding that?

**Comment by ahmed-mez** [comment]
Hey, looks like there is no intention to merge this PR so I'm closing it. Feel free to reopen if needed. Thanks!

**Pull request closed without merging**